### PR TITLE
[MI-2872]:Fixed issue #548 on github 'org lock for username'

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
https://github.com/mattermost/mattermost-plugin-github/issues/548

### CODE REVIEW

File: server/plugin/api.go [hunk 1]
```diff
@@ -1203,19 +1203,48 @@ func (p *Plugin) getMilestones(c *UserContext, w http.ResponseWriter, r *http.Re
 	p.writeJSON(w, allMilestones)
 }
 
+func getRepositoryList(c *UserContext, userName string, githubClient *github.Client, allRepos []*github.Repository, opt github.ListOptions) ([]*github.Repository, error) {
+	for {
+		repos, resp, err := githubClient.Repositories.List(c.Ctx, userName, &github.RepositoryListOptions{ListOptions: opt})
+		if err != nil {
+			return nil, err
+		}
+		allRepos = append(allRepos, repos...)
```
Comment (ayusht2810) [inline server/plugin/api.go]:
```suggestion
		}
		
		allRepos = append(allRepos, repos...)
```

File: server/plugin/api.go [hunk 2]
```diff
@@ -1203,19 +1203,48 @@ func (p *Plugin) getMilestones(c *UserContext, w http.ResponseWriter, r *http.Re
...
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
```
Comment (ayusht2810) [inline server/plugin/api.go]:
```suggestion
		}
		
		opt.Page = resp.NextPage
```

File: server/plugin/api.go [hunk 3]
```diff
@@ -1203,19 +1203,48 @@ func (p *Plugin) getMilestones(c *UserContext, w http.ResponseWriter, r *http.Re
...
+			p.writeAPIError(w, &APIErrorResponse{Message: "Failed to fetch repositories", StatusCode: http.StatusInternalServerError})
+		}
+	} else {
 		for {
-			repos, resp, err := githubClient.Repositories.List(c.Ctx, "", &github.RepositoryListOptions{ListOptions: opt})
-			if err != nil {
-				c.Log.WithError(err).Warnf("Failed to list repositories")
+			repos, resp, lErr := githubClient.Repositories.ListByOrg(c.Ctx, org, &github.RepositoryListByOrgOptions{Sort: "full_name", ListOptions: opt})
+			if lErr != nil {
+				if resp.StatusCode == 404 {
```
Comment (ayusht2810) [inline server/plugin/api.go]:
```suggestion
				if resp.StatusCode == http.StatusNotFound {
```

File: server/plugin/api.go [hunk 4]
```diff
@@ -1203,19 +1203,48 @@ func (p *Plugin) getMilestones(c *UserContext, w http.ResponseWriter, r *http.Re
...
+					invalidOrgName = true
+					break
+				}
+				c.Log.WithError(lErr).Warnf("Failed to list repositories by org")
```
Comment (raghavaggarwal2308) [inline server/plugin/api.go]:
```suggestion

				c.Log.WithError(lErr).Warnf("Failed to list repositories by organization")
```

File: server/plugin/api.go [hunk 5]
```diff
@@ -1203,19 +1203,48 @@ func (p *Plugin) getMilestones(c *UserContext, w http.ResponseWriter, r *http.Re
...
+		if err != nil {
+			c.Log.WithError(err).Warnf("Failed to list repositories")
+			p.writeAPIError(w, &APIErrorResponse{Message: "Failed to fetch repositories", StatusCode: http.StatusInternalServerError})
+		}
+	} else {
 		for {
-			repos, resp, err := githubClient.Repositories.List(c.Ctx, "", &github.RepositoryListOptions{ListOptions: opt})
-			if err != nil {
-				c.Log.WithError(err).Warnf("Failed to list repositories")
+			repos, resp, lErr := githubClient.Repositories.ListByOrg(c.Ctx, org, &github.RepositoryListByOrgOptions{Sort: "full_name", ListOptions: opt})
```
Comment (raghavaggarwal2308) [inline server/plugin/api.go:1242]: Constant for `full_name`?

File: server/plugin/api.go [hunk 6]
```diff
@@ -1203,19 +1203,48 @@ func (p *Plugin) getMilestones(c *UserContext, w http.ResponseWriter, r *http.Re
...
+}
+
 func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.Request) {
 	githubClient := p.githubConnectUser(c.Context.Ctx, c.GHInfo)
 
 	org := p.getConfiguration().GitHubOrg
 
 	var allRepos []*github.Repository
+	var err error
 	opt := github.ListOptions{PerPage: 50}
```
Comment (raghavaggarwal2308) [inline server/plugin/api.go]:
```suggestion
```

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/plugin/api.go (hunk 5 above)
Comment (Kshitij-Katiyar) [reply server/plugin/api.go:1242]: copied from above

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/plugin/api.go [hunk 7]
```diff
@@ -1203,19 +1203,50 @@ func (p *Plugin) getMilestones(c *UserContext, w http.ResponseWriter, r *http.Re
...
 		for {
-			repos, resp, err := githubClient.Repositories.List(c.Ctx, "", &github.RepositoryListOptions{ListOptions: opt})
-			if err != nil {
-				c.Log.WithError(err).Warnf("Failed to list repositories")
+			repos, resp, lErr := githubClient.Repositories.ListByOrg(c.Ctx, org, &github.RepositoryListByOrgOptions{Sort: "full_name", ListOptions: opt})
+			if lErr != nil {
+				if resp.StatusCode == http.StatusNotFound {
+					invalidOrgName = true
+					break
+				}
```
Comment (avas27JTG) [inline server/plugin/api.go:1247]: I think `404` is shown even when a repo is not accessible but the org/repo name is valid, see if this also needs to be handled here

Comment (Kshitij-Katiyar) [reply server/plugin/api.go:1247]: yeah, Abhishek, I tried with not accessible org as well but the error in both cases is the same and there is no defining difference for which we can differentiate between the 2 cases. As the error is also same there will be no difference for the user experience

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**